httpx/http_service.go

```diff
@@ -126,7 +149,8 @@ func (t *RetryTransport) RoundTrip(ctx context.Context, req *http.Request) (*htt
 		if resp.StatusCode >= 500 {
 			return nil, &RetryableHTTPError{Path: req.URL.String(), StatusCode: resp.StatusCode}
```

Comment:
What's the behavior of a request that retries but never succeeds? Does it still return a nil error? (looks like we don't have a test case for that [here](https://github.com/remind101/pkg/blob/httpservice-customizability/httpx/http_service_test.go#L60-L73))